packages/components/ContextualMenu/src/ContextualMenuItem.tsx

```diff
@@ -74,38 +76,33 @@ export const ContextualMenuItem = compose<ContextualMenuItemType>({
       icon: !!icon,
     };
 
-    /*
+    /**
      * On Desktop, focus gets moved to the root of the menu, so hovering off the menu does not automatically call onBlur as we expect it to.
-     * OnMouseEnter and onMouseLeave are overridden with the below callbacks that calls onFocus and onBlur explicitly
+     * onMouseLeave is overridden to explicitly call onBlur to simulate removing focus
+     * To achieve this, we override the onMouseLEave handler returned by useAsPressable, and replace it with our own. Inside our own
+     * onMouseLeave handler, we call useAsPressable's onMouseLEave handler,
      */
-    const onMouseEnter = React.useCallback(
-      (e) => {
-        pressable.props.onMouseEnter && pressable.props.onMouseEnter(e);
-        pressable.props.onFocus && pressable.props.onFocus(e);
-        e.stopPropagation();
-      },
-      [pressable],
-    );
-
-    const onMouseLeave = React.useCallback(
+    const { onMouseLeave, ...restPressableProps } = pressable.props;
+    const onMouseLeaveModified = React.useCallback(
       (e) => {
-        pressable.props.onMouseLeave && pressable.props.onMouseLeave(e);
-        pressable.props.onBlur && pressable.props.onBlur(e);
-        e.stopPropagation();
+        pressablePropsModified.onBlur(e);
+        onMouseLeave && onMouseLeave(e);
       },
-      [pressable],
+      [onMouseLeave],
```

Comment:
Does this need to track pressablePropsModified.onBlur, too?

Reply:
Yes, thanks!